src/io/http/src/main/scala/HTTPSource.scala

```diff
@@ -95,9 +95,10 @@ class HTTPSource(name: String, host: String, port: Int, sqlContext: SQLContext)
       val sliceEnd = endOrdinal - lastOffsetCommitted.offset.toInt - 1
       requests.slice(sliceStart, sliceEnd).map{ case(id, request) =>
         val row = new GenericInternalRow(2)
-        val idRow = new GenericInternalRow(2)
-        idRow.update(0, UTF8String.fromString(id.toString))
-        idRow.update(1, null)
+        val idRow = new GenericInternalRow(3)
+        idRow.update(0, null)
+        idRow.update(1, UTF8String.fromString(id.toString))
```

Comment:
UTF8String.fromString(id.toString) looks expensive. what type is id?

Reply:
its a long, for some reason Spark's internal row construction needs this UTF8 string instead of a regular string. The HTTP Source is mainly good for debugging though and will be deprecated for v2 stuff soon though